lib/charms/loki_k8s/v0/loki.py

```diff
@@ -139,3 +475,55 @@ def loki_push_api(self):
             Loki Push API endpoint
         """
         return self._stored.loki_push_api
+
+    @property
+    def _labeled_alert_groups(self) -> list:
+        """Load alert rules from rule files.
+
+        All rules from files for a consumer charm are loaded into a single
+        group. The generated name of this group includes Juju topology
+        prefixes.
+
+        Returns:
+            a list of Loki alert rule groups.
+        """
+        alerts = []
+        logger.warning(self._ALERT_RULES_PATH)
+        for path in Path(self._ALERT_RULES_PATH).glob("*.rule"):
+            if not path.is_file():
+                continue
+
+            logger.debug("Reading alert rule from %s", path)
+            with path.open() as rule_file:
+                # Load a list of rules from file then add labels and filters
+                try:
+                    rule = yaml.safe_load(rule_file)
+                    rule = self._label_alert_topology(rule)
+                    rule = self._label_alert_expression(rule)
+                    alerts.append(rule)
+                except Exception as e:
+                    logger.error("Failed to read alert rules from %s: %s", path.name, str(e))
```

Comment:
Not sure if that is what you meant:
```suggestion
                try:
                    rule = yaml.safe_load(rule_file)
                except Exception as e:
                    logger.error("Failed to read alert rules from %s: %s", path.name, str(e))
                else:
                    rule = self._label_alert_topology(rule)
                    rule = self._label_alert_expression(rule)
                    alerts.append(rule)
```